Kuadrant/limitador-operator
# enhance error handling and status reconciliation

### what 

* enhance error handling: reconciliation error are being exposed in the status section of the CR
* enhance status reconciliation

### verification steps

dev setup

```
make local-setup
```
deploy limitador with a redis secret reference that does not exist
```yaml
k apply -f - <<EOF
---
apiVersion: limitador.kuadrant.io/v1alpha1
kind: Limitador
metadata:
  name: limitador-sample
spec:
  storage:
    redis:
      configSecretRef:
        name: not-existing
  listener:
    http:
      port: 8080
    grpc:
      port: 8081
  limits:
    - conditions: ["get_toy == 'yes'"]
      max_value: 2
      namespace: toystore-app
      seconds: 30
      variables: []
EOF
```
Check the status 
```yaml
k get limitadors.limitador.kuadrant.io limitador-sample -o jsonpath='{.status}' | yq e -P
conditions:
  - lastTransitionTime: "2022-10-07T14:22:00Z"
    message: Secret "not-existing" not found
    reason: ReconcilliationError
    status: "False"
    type: Ready
observedGeneration: 1
service:
  host: limitador-limitador-sample.default.svc.cluster.local
  ports:
    grpc: 8081
    http: 8080
```
